manifests/otel-agent-reconciler-cm.yaml

```diff
@@ -0,0 +1,69 @@
+# Copyright 2022 Google LLC
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+apiVersion: v1
+kind: ConfigMap
+metadata:
+  name: otel-agent-reconciler
+  namespace: config-management-system
+  labels:
+    app: opentelemetry
+    component: otel-agent
+    configmanagement.gke.io/system: "true"
+    configmanagement.gke.io/arch: "csmr"
+data:
+  otel-agent-reconciler-config.yaml: |
+    receivers:
+      opencensus:
+    exporters:
+      opencensus:
+        endpoint: otel-collector.config-management-monitoring:55678
+        tls:
+          insecure: true
```

Comment:
What is the justification for disabling TLS?

Reply:
This exporter in otel-agent sends data to the otel-collector receivers that runs on the same local host, so TLS is likely not necessary.

Reply:
I'm not super familiar with these configurations - but it looks like the endpoint referenced in this block is for another Service running in the cluster, which may not be on the same local host. Am I missing something?

Reply:
This looks like the sidecar agent config for the exporter sending to the otel-collector service/deployment.

That said, does the otel-collector have a TLS cert generated and injected or is it just using http?

Reply:
If it's supposed to be the opencensus config for the reconciler writing to the sidecar agent, then the endpoint it wrong. It should be localhost. Otherwise, you're bypassing the agent.

Reply:
Ah yeah sorry, it's sending to a different pod. In this case
- our metrics are custom metric data which are not critical to the healthiness of k8s or our product
- encryption may introduce extra overhead, and the cert configuration is an extra step in UX (or the certs need to be shipped as well)
- (this part I'm not sure) e2e tests are verifying metrics for every sync operation, the encryption may add work to the verification

Reply:
FWIW, it looks like this section is just a copy of the existing agent config, which is already exporting metrics with TLS disabled.

IMO, we should probably enable TLS, but that would be a different feature/project.

So what we have here is fine for now.

Reply:
A copy it is indeed :]